sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorTraceExporter.java

```diff
@@ -568,25 +598,66 @@ private void exportRequest(SpanData span, List<TelemetryItem> telemetryItems) {
 
         data.setSource(getSource(attributes));
 
-        if (isAzureQueue(attributes)) {
-            // TODO (trask): for batch consumer, enqueuedTime should be the average of this attribute
-            //  across all links
-            Long enqueuedTime = attributes.get(AZURE_SDK_ENQUEUED_TIME);
-            if (enqueuedTime != null) {
-                long timeSinceEnqueued =
-                    NANOSECONDS.toMillis(span.getStartEpochNanos()) - SECONDS.toMillis(enqueuedTime);
-                if (timeSinceEnqueued < 0) {
-                    timeSinceEnqueued = 0;
-                }
-                if (data.getMeasurements() == null) {
-                    data.setMeasurements(new HashMap<>());
-                }
-                data.getMeasurements().put("timeSinceEnqueued", (double) timeSinceEnqueued);
+        // TODO (trask): for batch consumer, enqueuedTime should be the average of this attribute
+        //  across all links
+        Long enqueuedTime = attributes.get(AZURE_SDK_ENQUEUED_TIME);
+        if (enqueuedTime != null) {
+            long timeSinceEnqueuedMillis =
+                Math.max(
+                    0L, NANOSECONDS.toMillis(span.getStartEpochNanos()) - SECONDS.toMillis(enqueuedTime));
+            if (data.getMeasurements() == null) {
+                data.setMeasurements(new HashMap<>());
+            }
+            data.getMeasurements().put("timeSinceEnqueued", (double) timeSinceEnqueuedMillis);
+        }
+        Long timeSinceEnqueuedMillis = attributes.get(KAFKA_RECORD_QUEUE_TIME_MS);
+        if (timeSinceEnqueuedMillis != null) {
+            if (data.getMeasurements() == null) {
+                data.setMeasurements(new HashMap<>());
             }
+            data.getMeasurements().put("timeSinceEnqueued", (double) timeSinceEnqueuedMillis);
```

Comment:
looks like we are setting timeSinceEnqueued twice? if attributes.get(KAFKA_RECORD_QUEUE_TIME_MS) is not null, the code is overwriting the value. Does KAFKA_RECORD_QUEUE_TIME_MS have higher precedence?

Reply:
one is sent by the Azure SDK instrumentation, the other is sent by Kafka instrumentation, so there's no overlap